lint.sh

```diff
@@ -1,56 +1,89 @@
 #!/usr/bin/env bash
-set -e
-source ./test/scripts/test_runner_setup.sh
-
-SHOULD_FIX=false
-while getopts f flag
-do
-    case "${flag}" in
-        f) SHOULD_FIX=true;;
-        *)
-    esac
+
+[ -f "local.env" ] || touch local.env
+source local.env
+
+# Default values
+path="."
+fix=false
+prettier=false
+exit_code=0
+
+# Function to display help message
+usage() {
+  echo "Usage: lint.sh <path> [--fix, -f] [--prettier]"
+  exit 1
+}
+
+# Error handler to capture errors
+error_handler() {
+  # shellcheck disable=SC2317
+  exit_code=1
+}
+
+# Trap ERR to handle errors
+trap 'error_handler' ERR
+
+# Parse command-line arguments
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --fix|-f)
+      fix=true
+      shift
+      ;;
+    --prettier)
+      prettier=true
+      shift
+      ;;
+    *)
+      path="$1"
+      shift
+      ;;
+  esac
 done
 
-if $SHOULD_FIX
-then
-  echo "Fixing Linting and Formatting Issues"
-  NPM_JS_COMMAND="lint:fix-js"
-  NPM_YML_COMMAND="lint:fix-yml"
-  NEWLINE_FLAGS="--fix"
-  BLACK_FLAGS=""
-else
-  NPM_JS_COMMAND="lint-js"
-  NPM_YML_COMMAND="lint-yml"
-  NEWLINE_FLAGS=""
-  BLACK_FLAGS="--check"
+# Ensure path is a directory
+if [[ ! -d $path ]]; then
+  echo "Error: Path '$path' does not exist or is not a directory."
+  usage
 fi
 
-# ℹ️ Linting Javascript test/**/*.cjs,js,json,mjs,ts
-$RUN_TEST_RUNNER_CMD "npm run $NPM_JS_COMMAND"
+# Linting JS, YAML, Python scripts, and general whitespace
+if $fix; then
+  echo "ℹ️ Fixing linting issues which can be fixed automatically"
 
-# ℹ️ Linting Markdown **/*.md
-if $SHOULD_FIX
-then
-  $RUN_TEST_RUNNER_CMD "npm run lint:fix-md"
-fi
+  if $prettier; then
```

Comment:
This brings back the thing Ricky had setup with running prettier then linting, but makes it optional. I think this is a good compromise. The result will always be something that passes our eslint rules, and with prettier the code will be in some places--well--prettier (which I like).